cmd/state/exec3/state.go

```diff
@@ -116,15 +116,15 @@ func (rw *Worker) ResetState(rs *state.StateV3, accumulator *shards.Accumulator)
 	rw.stateWriter = state.NewStateWriterV3(rs, accumulator)
 }
 
-func (rw *Worker) Tx() kv.Tx        { return rw.chainTx }
-func (rw *Worker) DiscardReadList() { rw.stateReader.DiscardReadList() }
+func (rw *Worker) Tx() kv.TemporalTx { return rw.chainTx }
+func (rw *Worker) DiscardReadList()  { rw.stateReader.DiscardReadList() }
 func (rw *Worker) ResetTx(chainTx kv.Tx) {
```

Comment:
makes sense to accept TemporalTx here too

Reply:
it's rabbit hole - better do by another PR